zndraw/server/events.py

```diff
@@ -340,21 +365,43 @@ def scene_pause(data):
 @io.on("modifier:register")
 def modifier_register(data):
     data["token"] = session["token"]
+    if "modifier" not in app.config["PER-TOKEN-DATA"][session["token"]]:
+        # this is a dict for the pyclient, but it has to be for every webclient ...
+        app.config["PER-TOKEN-DATA"][session["token"]]["modifier"] = {}
 
     try:
         # we can only register one modifier at a time
         name = data["modifiers"][0]["name"]
-        if name in app.config["MODIFIER"]:
-            # issue with the same modifier name on different webclients / tokens!
-            # only for default we need to ensure, there is only one.
-            raise ValueError(f"Modifier {name} is already registered.")
-        app.config["MODIFIER"][name] = request.sid
+        if name in app.config["MODIFIER"]["default_schema"]:
+            msg = f"Modifier {name} is already registered (default)."
+            log.critical(msg)
+            emit("message:log", msg, to=request.sid)
+
+        if name in app.config["PER-TOKEN-DATA"][session["token"]]["modifier"]:
+            msg = f"Modifier {name} is already registered."
+            log.critical(msg)
+            emit("message:log", msg, to=request.sid)
```

Comment:
I wonder if the default behaviour should be to overwrite an already registered modifier? Say, I spot a bug in SiMGen, then I could change the code there, and reconnect to the same session with an updated version. Right now this will prohibit that.

Reply:
I'd keep the restricted functionality for now, but I'm open for a `register_modifer(..., overwrite=True)` to overwrite this in a dedicated PR. I'll make this an issue